### ISSUE TITLE
issue #95 and tried to apply the solutions recommended

### ISSUE THREAD

**Issue by yvester** [laurenz/oracle_fdw#403]
Hello,
I am struggling with the issue oracle_utils.c oracle_utils.c:22:17: fatal error: oci.h: No such file or directory 
I read the replies to #95 and I tried to apply the solutions recommended but it still not working for me.  Here is my config in case you see anything wrong. Thanks

```
[root@lpn4e95 client64]# cd /var/tmp/oracle_fdw-2.1.0
[root@lpn4e95 oracle_fdw-2.1.0]# make
gcc -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -DLINUX_OOM_SCORE_ADJ=0 -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fPIC -I/usr/include/oracle/19.5/client64/sdk/include -I/usr/include/oracle/19.5/client64/oci/include -I/usr/include/oracle/19.5/client64/rdbms/public -I/usr/include/oracle/12.2/client -I/usr/include/oracle/12.2/client64 -I/usr/include/oracle/12.1/client -I/usr/include/oracle/12.1/client64 -I/usr/include/oracle/11.2/client -I/usr/include/oracle/11.2/client64 -I/usr/include/oracle/11.1/client -I/usr/include/oracle/11.1/client64 -I/usr/include/oracle/10.2.0.5/client -I/usr/include/oracle/10.2.0.5/client64 -I/usr/include/oracle/10.2.0.4/client -I/usr/include/oracle/10.2.0.4/client64 -I/usr/include/oracle/10.2.0.3/client -I/usr/include/oracle/10.2.0.3/client64 -I. -I. -I/usr/include/pgsql/server -I/usr/include/pgsql/internal -D_GNU_SOURCE -I/usr/include/libxml2   -c -o oracle_utils.o oracle_utils.c
oracle_utils.c:22:17: fatal error: oci.h: No such file or directory
 #include <oci.h>
                 ^
compilation terminated.
make: *** [oracle_utils.o] Error 1
[root@lpn4e95 oracle_fdw-2.1.0]# echo $ORACLE_HOME
/usr/include/oracle/19.5/client64
[root@lpn4e95 oracle_fdw-2.1.0]# echo $LD_LIBRARY_PATH
/usr/include/oracle/19.5/client64
[root@lpn4e95 oracle_fdw-2.1.0]# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/usr/lib/oracle/19.5/client64:/usr/include/oracle/19.5/client64
[root@lpn4e95 oracle_fdw-2.1.0]# ls /usr/include/oracle/19.5/client64
ldap.h     occiAQ.h       occiData.h     oci1.h    ociapr.h  ocidfn.h   ocikpr.h    ocixstream.h  orid.h  oro.h
nzerror.h  occiCommon.h   occi.h         oci8dp.h  ocidef.h  ociextp.h  ociver.h    odci.h        ori.h   ort.h
nzt.h      occiControl.h  occiObjects.h  ociap.h   ocidem.h  oci.h      ocixmldb.h  oratypes.h    orl.h   xa.h
[root@lpn4e95 oracle_fdw-2.1.0]#
```

**Comment by laurenz** [comment]
Some remarks:

- Please, *never* build software as `root`. It's just too dangerous.

- It seems like you only installed the SDK files, but you will also need the "basic" or "basic light" package, else the linking step will fail.

The error you see is because

1. The `Makefile` doesn't specifically cater for 19.5.

2. I forgot to add `$ORACLE_HOME` to the include path.

I have just fixed 2. with 5ab7a7775f, so the easiest thing for you is to use [the current development version](/laurenz/oracle_fdw/archive/master.zip) of oracle_fdw rather than 2.1.0.

Perhaps it would also be good to upgrade to Oracle client 19.6 in the hope to get fewer Oracle bugs.

**Comment by yvester** [comment]
Hello @laurenz 
I applied your fix and upgraded to 19.6.  It now installs and works like a charm.
Thanks for the quick support, I appreciate it.
